packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -124,6 +133,22 @@ function textNodeContentsAreEqual(node: Node, vnode: VText, renderer: RendererAP
     return false;
 }
 
+function getAttrsToNotValidate(optOutStaticProp: string[] | true | undefined) {
+    if (isUndefined(optOutStaticProp)) {
+        return;
+    }
+    if (isTrue(optOutStaticProp)) {
+        return;
+    }
+    if (!Array.isArray(optOutStaticProp) || !optOutStaticProp.every((el) => isString(el))) {
```

Comment:
```suggestion
    if (!isArray(optOutStaticProp) || !optOutStaticProp.every((el) => isString(el))) {
```

We have an `isArray` in `@lwc/shared`. We should also import an `ArrayEvery` or create it if it doesn't exist.

Reply:
Addressing this was somewhat complex, due to the type of `Array.prototype.every` once extracted from the prototype. Hopefully the workaround will suffice.